### PR TITLE
Optimize map accesses in several places

### PR DESCRIPTION
**Pull request by rPraml**
In this PR I will try to optimize map access on several places and pick up @gbrail's comment: https://github.com/mozilla/rhino/pull/1575#discussion_r1730067620

### CODE REVIEW

File: rhino/src/main/java/org/mozilla/javascript/Hashtable.java [hunk 1]
```diff
@@ -123,20 +123,25 @@ public int size() {
...
-            } else {
-                last.next = nv;
-                nv.prev = last;
-                last = nv;
-            }
-        } else {
-            // Update the existing value and keep it in the same place in the list
-            map.get(nv).value = value;
-        }
+        map.compute(
```
Comment (rPraml) [inline rhino/src/main/java/org/mozilla/javascript/Hashtable.java:126]: we will save one hash computation, but have the price of an additional invocation of a lambda expression:
I did this short test:
```java
	@Test
	public void speed() {
		for (long i = 0L; i < 10_000_000L; i++) {
			ht.put("one", 1);
			ht.put("two", 2);
			ht.put("three", 3);
			ht.put("four", 4);
			ht.put("five", 5);
			ht.put("six", 6);
			ht.put("seven", 7);
			ht.put("eight", 8);
			ht.put("nine", 9);
			ht.put("ten", 10);

		}
	}
```
it runs with the old code in apporx 8,5sec and with the new code approx 5,5 sec

File: rhino/src/main/java/org/mozilla/javascript/JavaMembers.java [hunk 2]
```diff
@@ -595,23 +599,21 @@ private void reflect(
                     NativeJavaMethod setters = null;
                     String setterName = "set".concat(nameComponent);
 
-                    if (ht.containsKey(setterName)) {
```
Comment (rPraml) [inline rhino/src/main/java/org/mozilla/javascript/JavaMembers.java:598]: I think we can safely remove `containsKey` here, as we call `get` in the next line, which is checked for null by the `instanceOf` test.

File: rhino/src/main/java/org/mozilla/javascript/JavaMembers.java [hunk 3]
```diff
@@ -687,13 +689,11 @@ private Field[] getAccessibleFields(boolean includeProtected, boolean includePri
     private static MemberBox findGetter(
             boolean isStatic, Map<String, Object> ht, String prefix, String propertyName) {
         String getterName = prefix.concat(propertyName);
-        if (ht.containsKey(getterName)) {
-            // Check that the getter is a method.
-            Object member = ht.get(getterName);
```
Comment (rPraml) [inline rhino/src/main/java/org/mozilla/javascript/JavaMembers.java:692]: same here. Do not hit map twice

**Comment by gbrail** [comment]
Yes, this is a good change. Thanks!

**Pull request merged**